### PR TITLE
use os.IsPermission and os.IsNotExist in writeErr, to handle these errors properly

### PR DESCRIPTION
**Pull request by ilius**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -383,6 +384,12 @@ func writeErr(c ssh.Channel, id uint32, err error) error {
 		code = ssh_FX_OK
 	case io.EOF:
 		code = ssh_FX_EOF
+	default:
+		if os.IsPermission(err) {
+			code = ssh_FX_PERMISSION_DENIED
+		} else if os.IsNotExist(err) {
+			code = ssh_FX_NO_SUCH_FILE
+		}
 	}
```
Comment (dmitshur) [inline server.go]: You could consider taking the nested if/else if statement and merge it with the existing switch cases, like this:

```Go
switch {
case err == nil:
	code = ssh_FX_OK
case err == io.EOF:
	code = ssh_FX_EOF
case os.IsPermission(err):
	code = ssh_FX_PERMISSION_DENIED
case os.IsNotExist(err):
	code = ssh_FX_NO_SUCH_FILE
}
```

You could even move the `ssh_FX_FAILURE` default value into a `default:` case, so all the logic for selecting code is in one place.

**Comment by taruti** [comment]
Hello @ilius 

Like @shurcooL suggested, please refactor the switch. Thank you for the contribution, after that tweak I'll merge this PR.

**Comment by ilius** [comment]
@taruti done

File: server.go [hunk 2]
```diff
@@ -377,12 +378,18 @@ func writeErr(c ssh.Channel, id uint32, err error) error {
 	bs := make([]byte, len(failTmpl))
 	copy(bs, failTmpl)
 	binary.BigEndian.PutUint32(bs[5:], id)
-	var code ssh_fx = ssh_FX_FAILURE
+	var code ssh_fx
 	switch err {
```
Comment (dmitshur) [inline server.go]: This should become `switch {`.

File: server.go [hunk 3]
```diff
@@ -377,12 +378,18 @@ func writeErr(c ssh.Channel, id uint32, err error) error {
...
 	case nil:
```
Comment (dmitshur) [inline server.go]: `case err == nil:`.

File: server.go [hunk 4]
```diff
@@ -377,12 +378,18 @@ func writeErr(c ssh.Channel, id uint32, err error) error {
...
 		code = ssh_FX_OK
 	case io.EOF:
```
Comment (dmitshur) [inline server.go]: `case err == io.EOF:`.

**Review by dmitshur — CHANGES_REQUESTED** [review]
I guess this repo doesn't have CI, does it? The current version won't compile. You'll get errors like:

```
invalid case os.IsPermission(err) in switch on err (mismatched types bool and error)
invalid case os.IsNotExist(err) in switch on err (mismatched types bool and error)
```

**Comment by ilius** [comment]
Sorry, fixed

**Comment by taruti** [comment]
Thank you for the  contribution. Merged into master.

**Pull request merged**